app/services/analytics/builders.py

```
@@ -0,0 +1,447 @@
+        cwi_status_color = 'danger'
+        cwi_status_label = 'Critical'
+
+    # Build metrics list
+    metrics = []
+
+    # Participation Rate metric
+    if hasattr(snapshot_orm, 'participation_rate'):
+        participation_metric = build_metric_snapshot_view(
+            metric_name='Participation Rate',
+            current_value=Decimal(str(snapshot_orm.participation_rate)),
+            previous_value=Decimal(str(snapshot_orm.participation_trend == 'increasing' and 70 or 75)),
```

Ruff (0.16.1)

[warning] Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)